promql/parser/ast.go

```diff
@@ -119,12 +119,24 @@ type Call struct {
 	PosRange posrange.PositionRange
 }
 
+type MatrixSelectorBinOp struct {
```

Comment:
Feels like it could just be fields of matrix selector without new node.

Reply:
Not having a special node here would also make it forwards compatible if we wanted to just general binop here maybe?